.github/workflows/ci.yml

```diff
@@ -20,3 +20,16 @@ jobs:
           coverage: xdebug
       - run: composer install
       - run: vendor/bin/phpunit --coverage-text
+
+  static-analysis:
+    name: PHPStan
+    runs-on: ubuntu-20.04
+    continue-on-error: true
```

Comment:
Seems like an acceptable compromise at this point. Should we merge this as-is or do you plan to address the failures first and remove this?

FWIW: Not a big fan of how `continue-on-error` is implemented for GitHub actions, as it does properly mark the job as red, however the complete action status will also be marked red despite being successful. GitHub's UI shows this as a red status next to the commit (near the top of the page), but the status badge is still green.

Reply:
Removed it as it will succeed once the promise counterpart is merged and tagged.